pages/index.tsx

```diff
@@ -19,21 +34,136 @@ export default function Home(): ReactElement {
         <title>{t('title')}</title>
         <link rel="icon" href="/favicon.ico" />
       </Head>
-
       <Header />
       <Main />
       <Footer />
+      {console.dir({ claimData })} {/* @TODO: Remove. For development purposes only. */}
     </Container>
   )
 }
 
+export interface QueryParams {
+  user_key: string
+  uniqueNumber: string
+}
+
+function buildApiUrl(url: string, queryParams: QueryParams) {
+  const apiUrl = new URL(url)
+
+  for (const key in queryParams) {
+    apiUrl.searchParams.append(key, queryParams[key as 'user_key' | 'uniqueNumber'])
+  }
+
+  return apiUrl.toString()
+}
+
 export const getServerSideProps: GetServerSideProps = async ({ req, locale }) => {
   const isProd = process.env.NODE_ENV === 'production'
   const logger = isProd ? pino({}) : pino({ prettyPrint: true })
   logger.info(req)
 
+  /*
+   * Load environment variables to be used for authentication & API calls.
+   * @TODO: Handle error case where env vars are null or undefined.
+   */
+  // Request fields
+  const ID_HEADER_NAME: string = process.env.ID_HEADER_NAME ?? ''
+
+  // API fields
+  const API_URL: string = process.env.API_URL ?? ''
+  const API_USER_KEY: string = process.env.API_USER_KEY ?? ''
+
+  // TLS Certificate fields
+  const CERT_DIR: string = process.env.CERTIFICATE_DIR ?? ''
+  const P12_FILE: string = process.env.P12_FILE ?? ''
+  const P12_PATH: string = path.join(CERT_DIR, P12_FILE)
+
+  let apiData: JSON | null = null
+
+  // Returns certificate object with cert, key, and ca fields.
+  // https://dexus.github.io/pem/jsdoc/module-pem.html#.readPkcs12
+  async function getCertificate() {
+    const pemReadPkcs12 = promisify(pem.readPkcs12)
+    const pfx = fs.readFileSync(P12_PATH)
+
+    // TS does not play very nicely with util.promisify
+    // See, e.g., https://github.com/Microsoft/TypeScript/issues/26048
+    // Non-MVP TODO: Consider removing this ignore & TypeScriptifying.
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore -- TypeScript does not handle promisify well.
+    const keybundle = await pemReadPkcs12(pfx, {})
```

Comment:
i think the second parameter is optional, should we just omit it? or does having an empty object serve a purpose?

Reply:
Good call. I'll omit it.